### PR TITLE
Remove shapely dependency

### PR DESCRIPTION
**Pull request by phil-scale**
Removes shapely dependency from Nucleus repository.

[sc-357235](https://app.shortcut.com/scaleai/story/357235)

### CODE REVIEW

File: nucleus/metrics/geometry.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    def __repr__(self) -> str:
+        points = ", ".join([str(point) for point in self.points])
+        return f"GeometryPolygon({points})"
+
+
+Segment = Tuple[GeometryPoint, GeometryPoint]
+
+
+# alpha * a1 + (1 - alpha) * a2 = beta * b1 + (1 - beta) * b2
+def _segment_intersection(
```
Comment (ardila) [inline nucleus/metrics/geometry.py:66]: Did you test this out? Can you add a simple unit test for this logic?

**Review by ardila — APPROVED** [review]

File: nucleus/metrics/geometry.py (hunk 1 above)
Comment (phil-scale) [reply nucleus/metrics/geometry.py]: Added unit tests for all of the methods

**Comment by gatli** [comment]
Forgot to bump the version! Otherwise nice work 🙂

**Comment by phil-scale** [comment]
> Forgot to bump the version! Otherwise nice work 🙂

Actually I wanted to but there was no release branch made for 5.0.2, so I figured I'd lump them all together

**Pull request merged**